kusd/validator/validation_network.go

```diff
@@ -0,0 +1,135 @@
+package validator
+
+import (
+	"errors"
+	"fmt"
+	"github.com/kowala-tech/kUSD/accounts"
+	"github.com/kowala-tech/kUSD/accounts/abi/bind"
+	"github.com/kowala-tech/kUSD/common"
+	"github.com/kowala-tech/kUSD/contracts/network"
+	"github.com/kowala-tech/kUSD/core/types"
+	"math/big"
+)
+
+type ValidatorsChecksum [32]byte
+
+// ValidationNetwork is a gateway to validators contracts on the network
+type ValidationNetwork interface {
```

Comment:
By looking at this interface, I immediately thought about this one: https://github.com/kowala-tech/kUSD/blob/master/consensus/consensus.go#L35
There's a large part of the validator state (related to consensus) that eventually could be ported to our implementation of the tendermint consensus engine. Something like proof of activity protocol - https://github.com/ethereum/go-ethereum/tree/master/consensus/clique
From a testing point of view I'm not sure if it's the best solution.

Reply:
Not sure if I follow :|
But if we can port, and it's stable then we should.

This change actually allows to test the Validator without needing access to network contracts, thru it's interface

Reply:
It's something can be done later on (if necessary) - I will open an issue.